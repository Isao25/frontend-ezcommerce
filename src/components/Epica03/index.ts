import { PaginationComp } from "./paginationComponent";
export {
    PaginationComp
}
/* === DOCUMENTACIÓN AUTOMÁTICA AVANZADA ===
 * Archivo: index.ts
 * Ruta: 
 * Tamaño: 85 bytes
 * Líneas totales:        3
 * Líneas no vacías: 4
 * Caracteres:       85
 * Última actualización: 26/06/2025 05:58:49
 * 
 * 📦 IMPORTS:
 *   - { PaginationComp } from "./paginationComponent";
 * 
 * 🔀 MIDDLEWARES:
 *   - Express middleware functions: 0
 * 
 * 📊 ANÁLISIS DE DEPENDENCIAS:
 * 
 * 📈 MÉTRICAS DE COMPLEJIDAD:
 *   - Puntuación de complejidad: 0
 *   - Ratio código/comentarios: 4.00
 * 
 * 🤖 Generado automáticamente por TypeScript Advanced Documenter
 */
