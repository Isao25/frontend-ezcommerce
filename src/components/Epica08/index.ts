import { RecoverAccountModal } from "./RecoverAccountModal";
import { HelpAndSupport } from "./HelpAndSoport";

export{
    RecoverAccountModal,
    HelpAndSupport
}

/* === DOCUMENTACIÓN AUTOMÁTICA AVANZADA ===
 * Archivo: index.ts
 * Ruta: 
 * Tamaño: 166 bytes
 * Líneas totales:        7
 * Líneas no vacías: 6
 * Caracteres:      166
 * Última actualización: 26/06/2025 05:58:49
 * 
 * 📦 IMPORTS:
 *   - { HelpAndSupport } from "./HelpAndSoport";
 *   - { RecoverAccountModal } from "./RecoverAccountModal";
 * 
 * 🔀 MIDDLEWARES:
 *   - Express middleware functions: 0
 * 
 * 📊 ANÁLISIS DE DEPENDENCIAS:
 * 
 * 📈 MÉTRICAS DE COMPLEJIDAD:
 *   - Puntuación de complejidad: 0
 *   - Ratio código/comentarios: 6.00
 * 
 * 🤖 Generado automáticamente por TypeScript Advanced Documenter
 */
