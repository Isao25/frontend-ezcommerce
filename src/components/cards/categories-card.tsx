import { useNavigate } from "react-router";

interface ICategoriesCardProps {
    id:string,
    image:string,
    title:string,
    description:string,
    horiz:boolean,
    index:number
}

export const CategoriesCard = ({id, image, title, description, horiz, index }:ICategoriesCardProps) => {
    const navigate = useNavigate()
    const goToCategory = () => navigate(`/search?etiquetas=${id}`)

    const posicion = horiz ? `lg:col-span-2 lg:row-start-${index} lg:col-start-2` : `lg:row-span-2 lg:col-start-${index+1} lg:row-start-1`
    
    return (
      <button className={`${posicion} rounded-lg shadow-lg border-[1.5px] border-slate-400 p-4 flex lg:flex-${horiz ?'row':'col'} gap-4`}
        onClick={() => goToCategory()}>
        <div className={!horiz ? 'w-full':'w-full lg:w-1/2'}>
          <img
            src={image}
            className='w-full h-[180px] object-cover rounded-lg'
            alt={title}
          />
        </div>
        <div className="w-full">
          <h2 className='  sm:text-md md:text-xl lg:text-2xl xl:text-4xl font-bold mb-2 text-terciaryLight dark:text-terciaryDark'>{title}</h2>
          <hr className="mb-3"/>
          <p className='line-clamp-6'>{description}</p>
        </div>
      </button>
    );
  };