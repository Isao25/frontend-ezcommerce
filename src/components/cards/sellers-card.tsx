import { useNavigate } from 'react-router'
import  rightArrow  from '../../assets/icons/right-arrow.svg'
import  avatar  from '../../assets/avatar.png'
export interface ISellersCardProps {
    id:string,
    name:string,
}

export const SellersCard = (
    {
     id,
     name,
    }:ISellersCardProps) => {
        const navigate = useNavigate()
        const goToProfile = () => navigate(`/profile-buyer?id=${id}`)

        return(
            <>
                <div className="flex flex-col bg-white shadow-lg rounded-lg max-w-[280px] h-[330px] p-4">
                    <div className="rounded-lg overflow-hidden">
                        <img
                        src={avatar}
                        alt={`perfil-${id}`}
                        className='w-full h-[200px] object-cover'
                        />
                    </div>
                    <div className="mt-4">
                        <h3 className='text-xl truncate font-bold text-terciaryLight dark:text-terciaryDark'>
                            {name}
                        </h3>
                    </div>
                    <div className="flex mt-1">
                        <div className='justify-items-right pt-3' >
                            <button className='rounded-full bg-[#B7B7B7] bg-opacity-45 p-2'
                               
                            >
                                <img className='h-7 w-7' src={rightArrow} />
                            </button>
                        </div>
                    </div>
                </div>
            </>
        )
}

