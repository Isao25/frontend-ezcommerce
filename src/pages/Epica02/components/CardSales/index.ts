
/* === DOCUMENTACIÓN AUTOMÁTICA AVANZADA ===
 * Archivo: index.ts
 * Ruta: 
 * Tamaño: 0 bytes
 * Líneas totales:        0
 * Líneas no vacías: 0
 * Caracteres:        0
 * Última actualización: 26/06/2025 05:58:49
 * 
 * 🔀 MIDDLEWARES:
 *   - Express middleware functions: 0
 * 
 * 📊 ANÁLISIS DE DEPENDENCIAS:
 * 
 * 📈 MÉTRICAS DE COMPLEJIDAD:
 *   - Puntuación de complejidad: 0
 *   - Ratio código/comentarios: 0
 * 
 * 🤖 Generado automáticamente por TypeScript Advanced Documenter
 */
