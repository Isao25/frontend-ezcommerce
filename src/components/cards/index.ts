import { CategoriesCard } from "./categories-card";
import { SellersCard } from "./sellers-card";

export{
    SellersCard,
    CategoriesCard
}
/* === DOCUMENTACIÓN AUTOMÁTICA AVANZADA ===
 * Archivo: index.ts
 * Ruta: 
 * Tamaño: 144 bytes
 * Líneas totales:        6
 * Líneas no vacías: 6
 * Caracteres:      144
 * Última actualización: 26/06/2025 05:58:49
 * 
 * 📦 IMPORTS:
 *   - { CategoriesCard } from "./categories-card";
 *   - { SellersCard } from "./sellers-card";
 * 
 * 🔀 MIDDLEWARES:
 *   - Express middleware functions: 0
 * 
 * 📊 ANÁLISIS DE DEPENDENCIAS:
 * 
 * 📈 MÉTRICAS DE COMPLEJIDAD:
 *   - Puntuación de complejidad: 0
 *   - Ratio código/comentarios: 6.00
 * 
 * 🤖 Generado automáticamente por TypeScript Advanced Documenter
 */
