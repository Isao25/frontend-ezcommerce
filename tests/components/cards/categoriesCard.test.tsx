import { CategoriesCard } from '../../../src/components/cards';
import { render, screen, fireEvent } from '@testing-library/react';
import { ReactElement } from 'react';
import { BrowserRouter } from 'react-router-dom';

// Mock del hook useNavigate
const mockNavigate = jest.fn();
jest.mock('react-router', () => ({
  ...jest.requireActual('react-router'),
  useNavigate: () => mockNavigate,
}));

// Componente wrapper para las pruebas
export const renderWithRouter = (component: ReactElement) => {
  return render(<BrowserRouter>{component}</BrowserRouter>);
};

describe('CategoriesCard', () => {
  const defaultProps = {
    id: 'categoria-1',
    image: 'https://example.com/image.jpg',
    title: 'Título de Prueba',
    description: 'Esta es una descripción de prueba para el componente',
    horiz: false,
    index: 0
  };

  beforeEach(() => {
    mockNavigate.mockClear();
  });

  test('renderiza correctamente con props básicas', () => {
    renderWithRouter(<CategoriesCard {...defaultProps} />);
    
    expect(screen.getByText('Título de Prueba')).toBeInTheDocument();
    expect(screen.getByText('Esta es una descripción de prueba para el componente')).toBeInTheDocument();
    expect(screen.getByAltText('Título de Prueba')).toBeInTheDocument();
  });

  test('renderiza la imagen con src y alt correctos', () => {
    renderWithRouter(<CategoriesCard {...defaultProps} />);
    
    const image = screen.getByRole('img');
    expect(image).toHaveAttribute('src', 'https://example.com/image.jpg');
    expect(image).toHaveAttribute('alt', 'Título de Prueba');
  });

  test('navega correctamente cuando se hace clic', () => {
    renderWithRouter(<CategoriesCard {...defaultProps} />);
    
    const button = screen.getByRole('button');
    fireEvent.click(button);
    
    expect(mockNavigate).toHaveBeenCalledWith('/search?etiquetas=categoria-1');
    expect(mockNavigate).toHaveBeenCalledTimes(1);
  });

  test('aplica clases CSS correctas para layout horizontal', () => {
    const horizProps = { ...defaultProps, horiz: true, index: 1 };
    renderWithRouter(<CategoriesCard {...horizProps} />);
    
    const button = screen.getByRole('button');
    expect(button).toHaveClass('lg:col-span-2', 'lg:row-start-1', 'lg:col-start-2');
    expect(button).toHaveClass('lg:flex-row');
  });

  test('aplica clases CSS correctas para layout vertical', () => {
    const vertProps = { ...defaultProps, horiz: false, index: 2 };
    renderWithRouter(<CategoriesCard {...vertProps} />);
    
    const button = screen.getByRole('button');
    expect(button).toHaveClass('lg:row-span-2', 'lg:col-start-3', 'lg:row-start-1');
    expect(button).toHaveClass('lg:flex-col');
  });

  test('renderiza como un botón clickeable', () => {
    renderWithRouter(<CategoriesCard {...defaultProps} />);
    
    const button = screen.getByRole('button');
    expect(button).toBeInTheDocument();
    expect(button).toHaveClass('rounded-lg', 'shadow-lg', 'border-slate-400');
  });

  test('maneja diferentes índices correctamente', () => {
    const props1 = { ...defaultProps, horiz: false, index: 0 };
    const props2 = { ...defaultProps, horiz: false, index: 3 };
    
    const { rerender } = renderWithRouter(<CategoriesCard {...props1} />);
    let button = screen.getByRole('button');
    expect(button).toHaveClass('lg:col-start-1');
    
    rerender(<BrowserRouter><CategoriesCard {...props2} /></BrowserRouter>);
    button = screen.getByRole('button');
    expect(button).toHaveClass('lg:col-start-4');
  });

  test('maneja diferentes IDs para navegación', () => {
    const customProps = { ...defaultProps, id: 'categoria-especial' };
    renderWithRouter(<CategoriesCard {...customProps} />);
    
    const button = screen.getByRole('button');
    fireEvent.click(button);
    
    expect(mockNavigate).toHaveBeenCalledWith('/search?etiquetas=categoria-especial');
  });

  // Test de accesibilidad
  test('tiene la estructura de accesibilidad correcta', () => {
    renderWithRouter(<CategoriesCard {...defaultProps} />);
    
    // Debe tener un botón como elemento principal
    expect(screen.getByRole('button')).toBeInTheDocument();
    
    // La imagen debe tener texto alternativo
    const image = screen.getByRole('img');
    expect(image).toHaveAttribute('alt');
    expect(image.getAttribute('alt')).toBeTruthy();
  });

  // Test de snapshot para detectar cambios no intencionados
  test('coincide con el snapshot', () => {
    const { container } = renderWithRouter(<CategoriesCard {...defaultProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});