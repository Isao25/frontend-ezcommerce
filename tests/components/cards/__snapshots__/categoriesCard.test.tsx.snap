// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`CategoriesCard coincide con el snapshot 1`] = `
<button
  class="lg:row-span-2 lg:col-start-1 lg:row-start-1 rounded-lg shadow-lg border-[1.5px] border-slate-400 p-4 flex lg:flex-col gap-4"
>
  <div
    class="w-full"
  >
    <img
      alt="Título de Prueba"
      class="w-full h-[180px] object-cover rounded-lg"
      src="https://example.com/image.jpg"
    />
  </div>
  <div
    class="w-full"
  >
    <h2
      class="  sm:text-md md:text-xl lg:text-2xl xl:text-4xl font-bold mb-2 text-terciaryLight dark:text-terciaryDark"
    >
      Título de Prueba
    </h2>
    <hr
      class="mb-3"
    />
    <p
      class="line-clamp-6"
    >
      Esta es una descripción de prueba para el componente
    </p>
  </div>
</button>
`;
